pkg/utils/elasticsearch8.go

```diff
@@ -356,3 +369,123 @@ func (es *Elasticsearch8) DeleteTemplate(ctx context.Context, templateName strin
+		err = fmt.Errorf("pipeline cannot be deleted since pipeline %s is used by index %s", pipelineName, *pipelineStatus.index)
+		return err
+	}
+
+	response, err := esapi.IngestDeletePipelineRequest{PipelineID: pipelineName}.Do(ctx, es.Client)
+
+	if err != nil {
+		es.log.Error(err, "error while deleting pipeline", "pipelineName", pipelineName)
+		return err
+	}
+
+	defer response.Body.Close()
+
+	if !is2xxStatusCode(response.StatusCode) {
+		es.log.Error(nil, "error while deleting pipeline", "pipelineName", pipelineName, "http-response", response)
+		return fmt.Errorf("error while deleting pipeline %v: %v", pipelineName, response)
+	}
+
+	es.log.Info("pipeline was deleted successfully", "pipelineName", pipelineName)
+	return nil
+}
+
+func (es *Elasticsearch8) CreateOrUpdatePipeline(ctx context.Context, pipelineName string, model string) (*EsStatus, error) {
+	ctx, cancel := context.WithTimeout(ctx, ElasticMainFnTimeout)
+	defer cancel()
+
+	exists, err := es.existsPipeline(ctx, pipelineName)
+
+	if err != nil {
+		es.log.Error(err, "error while creating pipeline", "pipelineName", pipelineName)
+		return &EsStatus{Status: StatusError, Message: err.Error()}, err
+	}
+	response, err := esapi.IngestPutPipelineRequest{PipelineID: pipelineName, Body: strings.NewReader(model)}.Do(ctx, es.Client)
+	if err != nil {
+		es.log.Error(err, "error while creating pipeline", "pipelineName", pipelineName)
+		return &EsStatus{Status: StatusError, Message: err.Error()}, err
+	}
+
+	defer response.Body.Close()
+
+	if !is2xxStatusCode(response.StatusCode) {
+		es.log.Error(nil, "error while creating pipeline", "pipelineName", pipelineName, "http-response", response)
+		status := BuildEsStatus(response.StatusCode, response.String())
+		return status, errors.New("error while creating pipeline")
+	}
+
+	if exists {
+		es.log.Info("pipeline already exists and was updated successfully", "pipelineName", pipelineName)
+	} else {
+		es.log.Info("pipeline was created successfully", "pipelineName", pipelineName)
+	}
+
+	return BuildEsStatus(response.StatusCode, response.String()), nil
+}
+
+func (es *Elasticsearch8) isPipelineUsed(ctx context.Context, pipelineName string) (*PipelineStatus, error) {
+	req := esapi.IndicesGetSettingsRequest{}
+
+	res, err := req.Do(ctx, es.Client)
+	if err != nil {
+		es.log.Error(err, "Error retrieving indices settings")
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return nil, fmt.Errorf("error retrieving indices settings: %d", res.StatusCode)
+	}
+
+	var indicesSettings map[string]interface{}
+	if err := json.NewDecoder(res.Body).Decode(&indicesSettings); err != nil {
+		es.log.Error(err, "Error decoding indices elasticsearch response")
+		return nil, err
+	}
+	for indexName, indexSettings := range indicesSettings {
+		settings := indexSettings.(map[string]interface{})["settings"].(map[string]interface{})
+		defaultPipeline, found := settings["index"].(map[string]interface{})["default_pipeline"].(string)
```

Comment:
We could create constants for magic strings. Those constants can internal to the method.